### ISSUE TITLE
[DeepLabV3+] How to get class probabilities in C++ using frozen graph?

### ISSUE THREAD

**Issue by ZF-AR** [tensorflow/tensorflow#23814]
It looks like I should get the output from layer ResizeBilinear_3? This is the layer before argmax, according to tensorflow/models#4045. @aquariusjay
However, the numbers I got do not look like probabilities as they contained negative numbers and numbers much bigger than 1.
![image](https://user-images.githubusercontent.com/2267796/48654874-4af1f480-e9c5-11e8-9b20-215e144c3566.png)
FYI, the model I'm using is **deeplabv3_xception_ade20k_train** from the model zoo. It has 150 object classes + 1 unknown class so in total 151 classes.

Thank you!

**Comment by jtavrisov** [comment]
You need to use softmax to get a probability distribution. Softmax doesn't change order so it's irrelevant you are just going to argmax immediately but if you want real probabilities you need softmax.

**Comment by ZF-AR** [comment]
Indeed, it's from the output from layer ResizeBilinear_3 to get the class probabilities, albeit there are negative numbers and numbers much bigger than 1.

My followup question is whether we know the range, i.e. smallest negative value and largest positive value, in order to properly normalize? Or simply apply softmax?
@aquariusjay @jtavrisov 

Thank you!

**Comment by amaanda** [comment]
I am also working with the inner lays ResizeBilinear_3 and Argmax. @ZF-AR did you find out anything about the fixed range?

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
